railties/test/application/help_test.rb

```diff
@@ -21,5 +21,31 @@ def teardown
   test "short-cut alias works" do
     output = rails("-h")
     assert_match "The most common rails commands are", output
+    assert_match "In addition to those commands", output
+  end
+
+  test "help is the default command when no arguments are passed" do
+    output = rails("")
+    assert_match "The most common rails commands are", output
+    assert_no_match "In addition to those commands", output
+  end
+
+  test "outside application root it lists common commands only" do
+    output = gem_rails("")
+    assert_match "The most common rails commands are", output
+    assert_no_match "In addition to those commands", output
+  end
+
+  test "outside application root it lists common commands only with help option" do
+    output = gem_rails("-h")
+    assert_match "The most common rails commands are", output
+    assert_no_match "In addition to those commands", output
+  end
+
+  def gem_rails(cmd)
+    cmd = "#{Gem.ruby} #{RAILS_FRAMEWORK_ROOT}/railties/exe/rails #{cmd}"
+    output = `#{cmd}`
+    raise "Command #{cmd.inspect} failed. Output:\n#{output}" unless $?.success?
+    output
   end
```

Comment:
`gem_rails` should probably be a private method.  Also, I'm inclined to use `system` with the `exception: true` option, though I don't have a strong preference:

```suggestion
  private
    def gem_rails(cmd)
      capture(:stdout) do 
        system("#{Gem.ruby} #{RAILS_FRAMEWORK_ROOT}/railties/exe/rails #{cmd}", exception: true)
      end
    end
```